src/js/ia/IAPageNew.js

```diff
@@ -41,8 +41,93 @@
                     });
                 }
             });
+
+            $(".login_newia").click(function(evt) {
+                $("#signup-bg, #signup-form").addClass("hide");
+                $("#login-bg, #login-form").removeClass("hide");
+            });
+
+            $("#login-save").click(function(evt) {
+                var prefix = "#login-";
+                var username = $(prefix + "username-input").val();
+                var pwd = $(prefix + "pwd-input").val();
+                
+                $(".error-msg").addClass("hide");
+                
+                if (username.length && (pwd.length >= 8)) {
```

Comment:
We shouldn't so any password validation on login (or, arguably, on registration... but people complain when we don't) - that's what the auth layer is for.

Username validation needs to be the same as the [existing regex](https://github.com/duckduckgo/community-platform/blob/755037f/lib/DDGC/Web/Controller/My.pm#L716-L719) and include the same feedback on the form.

The username forms part of a XMPP JID. I'm not sure what the valid set is for that, but the existing validation has worked out so far.